### DISCUSSION TITLE
@micropython.asm_thumb on ESP32?

### DISCUSSION THREAD

**Discussion by Fisherman48** [micropython/micropython#11016] [ESP32] [answered]
Hi
@micropython.asm_thumb
 gives
'SyntaxError: invalid micropython decorator'

How get started with assembler on ESP32/ 8266 ??
( I'am used to asm on risc like Propeller bud not on esp familly)

BR /Stefan

**Comment by robert-hh** [comment, marked as answer]
There is no support for asm on ESP32 yet, and it is not an ARM architecture, it's XTENSA

**Reply by Fisherman48** [reply]
OK.
Not yet, but are there any plans for the future ?

I'm using the RMT for fast and accurate timing and I can read about SYNC of multiple channels in: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/rmt.html#multiple-channels-simultaneous-transmission,  but not in the esp32 module.
Are there any way to get info about plans/ work / expectations of future versions of upython?

BR /Stefan

**Comment by Rumidom** [comment]
Is the lack of asm support for the esp32 boards due to the XTENSA architecture? it would be a nice to have. Even if just for the RISC-V boards

**Reply by agatti** [reply]
All ESP32 boards based on RISC-V already have inline assembler support but it's disabled by default.  You may need to rebuild MicroPython for your specific board with `MICROPY_EMIT_INLINE_RV32` set to 1.

Xtensa LX6/LX7 has been in my TODO list for a while now - I guess I should get back to it :)

**Reply by Rumidom** [reply]
cool, I might do that some day
